api/src/Modules/staff/Controller/PutStaffMembership.php

```diff
@@ -0,0 +1,107 @@
+<?php declare(strict_types=1);
+/*.
+    require_module 'standard';
+.*/
+
+/**
+ *  @OA\Put(
+ *      tags={"members"},
+ *      path="/member/{id}/staff_membership",
+ *      summary="Update a staff membership for the member",
+ *      @OA\Parameter(
+ *          description="The ID of the member",
+ *          in="path",
+ *          name="id",
+ *          required=true,
+ *          @OA\Schema(
+ *              description="Member ID",
+ *              type="integer"
+ *          )
+ *      ),
+ *      @OA\RequestBody(
+ *          @OA\MediaType(
+ *              mediaType="application/x-www-form-urlencoded",
+ *              @OA\Schema(
+ *                  @OA\Property(
+ *                      property="Department",
+ *                      description="ID or name of the department",
+ *                      nullable=false,
+ *                      type="string"
+ *                  ),
+ *                  @OA\Property(
+ *                      property="Position",
+ *                      description="ID of the position",
+ *                      nullable=false,
+ *                      type="integer"
+ *                  ),
+ *                  @OA\Property(
+ *                      property="Note",
+ *                      description="Notes about member",
+ *                      nullable=true,
+ *                      type="string"
+ *                  )
+ *              )
+ *          )
+ *      ),
+ *      @OA\Response(
+ *          response=200,
+ *          description="OK"
+ *      ),
+ *      @OA\Response(
+ *          response=401,
+ *          ref="#/components/responses/401"
+ *      ),
+ *      security={{"ciab_auth":{}}}
+ *  )
+ */
+
+
+namespace App\Modules\staff\Controller;
+
+use Slim\Http\Request;
+use Slim\Http\Response;
+use App\Controller\InvalidParameterException;
+use Atlas\Query\Update;
+
+class PutStaffMembership extends BaseStaff
+{
+
+
+    public function buildResource(Request $request, Response $response, $params): array
+    {
+        $body = $request->getParsedBody();
+        if (!$body) {
+            throw new InvalidParameterException('Body required');
+        }
+
+        if (!array_key_exists('Department', $body)) {
+            throw new InvalidParameterException('Required \'Department\' parameter not present');
+        }
+
+        if (!array_key_exists('Position', $body)) {
+            throw new InvalidParameterException('Required \'Position\' parameter not present');
+        }
```

Comment:
There is a new function that can be called instead of a bunch of array_key_exists

`$request->checkRequiredBody`

See if that works for you. I have used it in the newer implementations but have not gone back and replaced it other places.

Reply:
I made the change here and it works great, thanks for the tip! Let me know if the changes look better to you.